lib/plugins/aws/provider.js

```diff
@@ -1698,6 +1744,170 @@ class AwsProvider {
+
+        const buildDockerArgs = ['build', '-t', `${repositoryName}:${imageName}`, imagePath];
+        await spawnExt('docker', buildDockerArgs, { stdio: 'inherit' });
+
+        const tagDockerArgs = [
+          'tag',
+          `${repositoryName}:${imageName}`,
+          `${repositoryUri}:${imageName}`,
+        ];
+        await spawnExt('docker', tagDockerArgs, { stdio: 'inherit' });
+
+        const pushDockerArgs = ['push', `${repositoryUri}:${imageName}`];
+        const dockerPushPromise = spawnExt('docker', pushDockerArgs);
+        dockerPushPromise.stdout.pipe(process.stdout);
+        dockerPushPromise.stderr.pipe(process.stderr);
+        await dockerPushPromise;
+
+        // Extract imageSha from `docker push` output
+        const imageSha = dockerPushPromise.stdoutBuffer
+          .toString()
+          .match(/(sha256:[a-f0-9]{64})/)[0];
+
+        return {
+          functionImageSha: imageSha.slice('sha256:'.length),
+          functionImageUri: `${repositoryUri}@${imageSha}`,
+        };
+      },
+
+      { promise: true }
+    ),
+    resolveImageUriAndShaFromUri: d(
+      async function (image) {
+        const providedImageSha = image.split('@')[1];
+        if (providedImageSha) {
+          return {
+            functionImageSha: providedImageSha.slice('sha256:'.length),
+            functionImageUri: image,
+          };
+        }
+
+        const [repositoryName, imageTag] = image.split('/')[1].split(':');
+        const registryId = image.split('.')[0];
+        const describeImagesResponse = await this.request('ECR', 'describeImages', {
+          imageIds: [
+            {
+              imageTag,
+            },
+          ],
+          repositoryName,
+          registryId,
+        });
+        const imageDigest = describeImagesResponse.imageDetails[0].imageDigest;
+        const functionImageUri = `${image.split(':')[0]}@${imageDigest}`;
+        return {
+          functionImageUri,
+          functionImageSha: imageDigest.slice('sha256:'.length),
+        };
+      },
+      { promise: true }
+    ),
+    resolveImageUriAndSha: d(
+      // TODO: TESTS
+      async function (image) {
+        // TODO: Add better validation here
+        const imageDefinedInProvider = this.serverless.service.provider.images[image];
+        if (imageDefinedInProvider) {
+          if (_.isObject(imageDefinedInProvider)) {
+            if (imageDefinedInProvider.path) {
+              return await this.resolveImageUriAndShaFromPath(image, imageDefinedInProvider.path);
+            }
+            return await this.resolveImageUriAndShaFromUri(imageDefinedInProvider.uri);
+          }
+          if (isDockerUri(imageDefinedInProvider)) {
+            return await this.resolveImageUriAndShaFromUri(imageDefinedInProvider);
+          }
+          return await this.resolveImageUriAndShaFromPath(image, imageDefinedInProvider);
+        }
+        return await this.resolveImageUriAndShaFromUri(image);
```

Comment:
Instaed of blindly assuming it's an `image` uri, We should probably confirm dev doesn't try to address  image supposedly set at `provider.images` (e.g. typo can be introduced, and assuming URI, error could be confusing)

```javascript
if (!isDockerUri(imageDefinedInProvider)) {
 throw `${image} not defined in `provider.images`;
}

Reply:
Yeah, that's definitely a thing to improve, I even added a vague TODO comment on line 1892 to address that specific issue :+1: Good catch :+1: